src/ab-test-preview.js

```diff
@@ -0,0 +1,44 @@
+import { getLetter } from '../src/utils';
+
+const queryString = window.location.search;
+const urlParams = new URLSearchParams(queryString);
+const abtest = urlParams.get( 'set_test' );
+const abIndex = parseFloat( urlParams.get( 'i' ) );
+
+document.querySelectorAll( '.ab-test-xb-preview' ).forEach( function ( xb ) {
+
+	const templates = document.querySelectorAll( 'template[data-parent-id="' + xb.dataset.clientId + '"]' );
+	const tabContainer = xb.querySelector( '.ab-test-xb-preview__tabs' );
+	const tabContent = xb.querySelector( '.ab-test-xb-preview__content' );
+
+	for ( let i = 0; i < templates.length; i++ ) {
+
+		const tab = document.createElement( 'button' );
+		if ( i === 0 || i === abIndex && xb.dataset.clientId === abtest ){
```

Comment:
I still think it's worth avoiding the duplication with the click handler code and handling this logic to set the default tab after looping over the templates. You have all the info you need to select the right tab and call `.click()` on it as it'll either be index 0 or the value of `abIndex` if the block client ID matches.

Reply:
In my latest commit ( I just did a commit now ) I made some further changes to the click handler